### PR TITLE
Rhb 4

### PR DESCRIPTION
**Pull request by Avaxapsys**
Didn't publish SP to DB yet, couse don't want to clean others

### CODE REVIEW

File: DevEdu.API/DevEdu.DAL/Models/CourseDto.cs [hunk 1]
```diff
@@ -6,7 +6,7 @@ public class CourseDto : BaseDto
     {
         public string Name { get; set; }
         public string Description { get; set; }
-        public List<CourseTopicDto> CourseTopics { get; set; }
+        public List<TopicDto> Topics { get; set; }
```
Comment (Negodyaj) [inline DevEdu.API/DevEdu.DAL/Models/CourseDto.cs:9]: CourseTopicDto is not a simple relational entity, it has Position inside
so you can either use CourseTopicDto here or implement ordering inside the stored procedure
but anyway, Position of each topic must be returned from controller somehow =)

File: DevEdu.API/DevEdu.DAL/Repositories/CourseRepository.cs [hunk 2]
```diff
@@ -45,11 +46,36 @@ public void DeleteCourse(int id)
 
         public CourseDto GetCourse(int id)
         {
-            return _connection.QuerySingleOrDefault<CourseDto>(
+            return _connection.Query<CourseDto, TopicDto, MaterialDto, TaskDto, GroupDto, CourseDto>(
```
Comment (Negodyaj) [inline DevEdu.API/DevEdu.DAL/Repositories/CourseRepository.cs:49]: do we really need all these entities at the same time?)

File: DevEdu.API/DevEdu.DAL/Repositories/CourseRepository.cs [hunk 3]
```diff
@@ -45,11 +46,36 @@ public void DeleteCourse(int id)
...
+                    else
+                        course.Groups.Add(group);
+
+                    return course;
+                },
                 new { id },
+                splitOn: "Id",
                 commandType: CommandType.StoredProcedure
-            );
+            ).FirstOrDefault();
```
Comment (Negodyaj) [inline DevEdu.API/DevEdu.DAL/Repositories/CourseRepository.cs:78]: and please check the result of this method on some real data

File: DevEdu.API/DevEdu.DAL/Models/CourseDto.cs (hunk 1 above)
Comment (Avaxapsys) [reply DevEdu.API/DevEdu.DAL/Models/CourseDto.cs:9]: Add to TopicDto

File: DevEdu.API/DevEdu.DAL/Repositories/CourseRepository.cs (hunk 2 above)
Comment (Avaxapsys) [reply DevEdu.API/DevEdu.DAL/Repositories/CourseRepository.cs:49]: It'll be next task

Comment (Avaxapsys) [reply DevEdu.API/DevEdu.DAL/Repositories/CourseRepository.cs]: I've kept only group in related, couse topic, material and task have more common sense

File: DevEdu.API/DevEdu.Db/StoredProcedures/Course/Course_SelectById.sql [hunk 4]
```diff
@@ -3,7 +3,15 @@
...
 BEGIN
 	SELECT 
-	Id, Name, Description, IsDeleted
-	FROM dbo.Course
-	WHERE (Id = @Id)
+	   C.Id
+      ,C.Name
+      ,C.Description
+      ,C.IsDeleted
+	  ,G.Timetable
```
Comment (Negodyaj) [inline DevEdu.API/DevEdu.Db/StoredProcedures/Course/Course_SelectById.sql:10]: g.timetable must be placed after g.id

**Pull request merged**